code/atoms/_atom.dm

```diff
@@ -10,13 +10,29 @@
 	var/self_move = FALSE
 	var/move_sound
 
+	var/shadow_size = null
+	var/shadow_pixel_x = 0
+	var/shadow_pixel_y = 0
+
 /atom/proc/update_icon(var/list/supplied = list())
 	overlays = supplied
 	var/mob/holder = loc
 	if(istype(holder))
 		holder.update_inventory()
 		holder.update_icon()
 
+/atom/movable/update_icon(var/list/supplied = list())
+	if((flags & FLAG_SIMULATED) && !isnull(shadow_size))
+		underlays.Cut()
```

Comment:
I should really build a framework for modifying underlays and overlays, so we aren't wiping and rebuilding them everywhere.